tests/test_bounds.sh

```diff
@@ -0,0 +1,21 @@
+#!/bin/sh
```

Comment:
We've found (from bitter experience) that it's a good idea to have `set -e` as the first line of shell scripts, as it forces the shell to stop executing as soon as an error is encountered.

Reply:
Yes, that's, of course, an excellent choice in general.

Reply:
From the Bash manual (emphasis mine):

> This option applies to the shell environment and each subshell environment separately (see Command Execution Environment), and **may cause subshells to exit before executing all the commands in the subshell**.

After running the SSH command, in fact, it exits.

We have two options here:

- Avoid `set -e`
- executing locally CheriBSD (the redirections could be an issue here?)

Reply:
Are you saying this *definitely* causes a problem or are you saying it *might* cause a problem?

[I use `set -e` with `ssh` extensively and have never observed an issue.]

Reply:
It is a problem actually. After the SSH command, exits. So, it can't run the rest to check the result.

Reply:
I don't understand what "after the SSH command, exits" means.

Reply:
But the problem may be the redirection, I suppose. And we need the redirection for these "raw" tests.

Reply:
> I don't understand what "after the SSH command, exits" means.

You are right. I'll be more clear.

The script runs until this line `RESULT_riscv64=`ssh -p $SSHPORT -t root@127.0.0.1 'sh -c "/root/bounds 2>&1 > /dev/null"'``.

Reply:
I still don't understand.

Reply:
> I still don't understand.

With `set -e`, all these lines are not executed:

`bash 
if [ -z "${RESULT_riscv64##*In-address space security exception (core dumped)*}" ] ;then
        echo "TEST PASSED!"
    else
        echo "TEST FAILED!"
        echo "Output: $RESULT_riscv64"
fi
else
    echo "Please, build first before running the tests."
    exit 1
fi
`

Reply:
Hang on, that line is *supposed* to fail, right? If so you need to do something like `{X && exit 1} || true`.

Reply:
Which line you are referring to?

Reply:
Oh the RESULT! Right.